File: src/main/java/pl/training/workshop/m5/s02_pullupfield/step3/VipTicket.java
package pl.training.workshop.m5.s02_pullupfield.step3;

import java.util.Locale;

/** Krok 3: reguła normalizacji VIP zostaje w podklasie - przekazuje do bazy już znormalizowaną wartość. */
public final class VipTicket extends Ticket {
    public VipTicket(String seat) {
        super(seat.toUpperCase(Locale.ROOT));
    }

    @Override
    public String describe() {
        return "VIP " + seat();
    }
}

File: typescript/src/workshop/m5/s02_pullupfield/step3/VipTicket.ts
import { Ticket } from './Ticket.js';

/** Krok 3: reguła normalizacji VIP zostaje w podklasie - przekazuje do bazy już znormalizowaną wartość. */
export class VipTicket extends Ticket {
  constructor(seat: string) {
    super(seat.toUpperCase());
  }

  override describe(): string {
    return 'VIP ' + this.seat();
  }
}
